pkg/scheduler/queue/leafqueue.go

```diff
@@ -0,0 +1,137 @@
+package queue
+
+import (
+	"fmt"
+	"strings"
+)
+
+type QueuePath []string //nolint:revive
+
+// LeafQueue is an hierarchical queue implementation where each sub-queue
+// has the same guarantees to be chosen from.
+// Each queue has also a local queue, which gets chosen from first. Only if the
+// local queue is empty, items from the sub-queues are dequeued.
+type LeafQueue struct {
+	// local queue
+	ch RequestChannel
+	// index of where this item is located in the mapping
+	pos QueueIndex
+	// index of the sub-queues
+	current QueueIndex
+	// mapping for sub-queues
+	mapping *Mapping[*LeafQueue]
+	// name of the queue
+	ident string
```

Comment:
nit: why not name it as `id` instead or just `name`? `ident` sometimes looks like `indent`. May be it's just me